Sources/CasePaths/EnumReflection.swift

```diff
@@ -69,22 +71,36 @@ public func extract<Root, Value>(case embed: @escaping (Value) -> Root, from roo
 public func extract<Root, Value>(_ embed: @escaping (Value) -> Root) -> (Root) -> (Value?) {
   var cachedTag: UInt32?
   return { root in
-    guard let rootTag = enumTag(root) else { return nil }
-    if let cachedTag = cachedTag, cachedTag != rootTag { return nil }
+    guard let metadata = EnumMetadata(Root.self) else { return nil }
+    let rootTag = metadata.tag(of: root)
+
+    if let cachedTag = cachedTag {
+      guard cachedTag == rootTag else { return nil }
+      if metadata.typeDescriptor.fieldDescriptor.field(atIndex: rootTag).isIndirectCase {
+        return .some(metadata.indirectAssociatedValue(of: root, as: Value.self))
+      } else {
+        return .some(metadata.directAssociatedValue(of: root, as: Value.self))
+      }
+    }
+
     guard
       let value = (Mirror(reflecting: root).children.first?.value)
         .flatMap({ $0 as? Value ?? Mirror(reflecting: $0).children.first?.value as? Value })
     else {
       #if compiler(<5.2)
         // https://bugs.swift.org/browse/SR-12044
-        if MemoryLayout<Value>.size == 0, !isUninhabitedEnum(Value.self) {
+        if
+          MemoryLayout<Value>.size == 0,
+          let valueMetadata = EnumMetadata(Value.self),
+          valueMetadata.kind == .enumeration,
```

Comment:
This can also be removed, right? Does `EnumMetadata` make that assumption?
```suggestion
```

Reply:
`EnumMetadata` checks that the metadata kind is either `.enumeration` or `.optional` (they use the same metadata format), but this workaround code (prior to my changes) checked specifically for the `.enumeration` kind.

I'm not entirely sure what checks are needed in this workaround and didn't want to download an older Xcode to play around with it.

Reply:
Since I had to download Xcode 11.3 to investigate the CI failure, I was able to test this and discover that I broke the workaround. It's fixed as of dd0a6d8.